uint32 class_id
uint32 instance_id
uint32 x
uint32 y
uint32 w
uint32 h